optaplanner-examples/src/main/java/org/optaplanner/examples/scrabble/optional/score/ScrabbleConstraintProvider.java

```diff
@@ -0,0 +1,67 @@
+package org.optaplanner.examples.scrabble.optional.score;
+
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintCollectors;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.api.score.stream.Joiners;
+import org.optaplanner.examples.scrabble.domain.ScrabbleWordDirection;
+import org.optaplanner.examples.scrabble.domain.ScrabbleCell;
+import org.optaplanner.examples.scrabble.domain.ScrabbleWordAssignment;
+
+public class ScrabbleConstraintProvider implements ConstraintProvider {
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[]{
+                                characterConflict(constraintFactory),
+                                noParallelHorizontalNeighbours(constraintFactory),
+                                noParallelVerticalNeighbours(constraintFactory),
+                                outOfGrid(constraintFactory),
+                                maximizeMergesPerWord(constraintFactory),
+                                pullToCenter(constraintFactory)
+        };
```

Comment:
One last thing, @Christopher-Chianelli, and then I'll merge: this isn't the proper code style, we don't do alignment on brackets. Please check the rest of the file as well, it's not the only place I see it.

Reply:
I use `KIE Java Conventions` in Eclipse. Fun fact: the IDEA `KIE Java Formatter` and Eclipse `KIE Java Formatter` disagrees in some places, and this is one of them. I remember I created a script once that ran Idea formatter for me from command line. Might be harder to do that now that I have IDEA in a flatpak....

Reply:
Dammit. :-( We should somehow get them in sync. I'll bring it up.
In the meantime - any chance you could configure yours to be the same as in IDEA? As far as I know, most of the team uses IntelliJ.

Reply:
They are fundamentally incompatible (IDEA's have more features than Eclipse). I will try running the IDEA's formatter in a container and package it as a program.

Reply:
Done; managed to find out where `format.sh` is in the flatpak.